addOns/oast/src/main/java/org/zaproxy/addon/oast/OastInsertPayloadHandler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast;
+
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import org.parosproxy.paros.Constant;
+
+@SuppressWarnings("serial")
+public class OastInsertPayloadHandler extends JMenu {
+    private ExtensionOast extension;
+    private JMenuItem boastMenuItem;
+    private JMenuItem interactshMenuItem;
+    private JMenuItem callbackMenuItem;
+    private OastInsertPayloadMenu parentMenu;
+    private boolean isNew;
+
+    public OastInsertPayloadHandler(
+            OastInsertPayloadMenu parentMenu, ExtensionOast extension, boolean isNew) {
+        super(Constant.messages.getString("oast.popup.menu." + (isNew ? "new" : "existing")));
+        this.parentMenu = parentMenu;
+        this.extension = extension;
+        this.isNew = isNew;
+        initialize();
+    }
+
+    public void initialize() {
+        boastMenuItem = new JMenuItem(Constant.messages.getString("oast.popup.menu.boast"));
+        interactshMenuItem =
+                new JMenuItem(Constant.messages.getString("oast.popup.menu.interactsh"));
+        callbackMenuItem = new JMenuItem(Constant.messages.getString("oast.popup.menu.callback"));
```

Comment:
The services should be handled uniformly, i.e. use `ExtensionOast#getServices()`. If tomorrow we add a new service we should not need to change this code to add another entry.